mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -2116,8 +2114,8 @@ impl TakerSwap {
             return ERR!("Taker payment is refunded, swap is not recoverable");
         }
 
-        if self.r().maker_payment_spend.is_some() {
-            return ERR!("Maker payment is spent, swap is not recoverable");
+        if self.r().maker_payment_spend.is_some() && self.r().maker_payment_spend_confirmed {
+            return ERR!("Maker payment is spent and confirmed, swap is not recoverable");
```

Comment:
yeah, that is the reason of not working recover funds. Thanks for fixing!

Reply:
This is what required a change of required confirmations as well, since if it was still 1 for polygon we could have cases where this error is returned but the transaction is not on chain.

Reply:
What if the payment spend tx is in mempool, so maker_payment_spend_confirmed is false but we cannot yet recover funds?

Reply:
If payment spend tx is in mempool then the confirmation step will never happen and swap is still in progress. It will not be shown as failed in GUI and the user will not have the recover funds button to recover funds. We should add RBF and CPFP in the future though bump the transaction.